tests/src/Integration/Action/SystemPageRedirectTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\SystemPageRedirectTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action {
+  use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+  use Psr\Log\LoggerInterface;
+  use Psr\Log\LogLevel;
+
+
+  /**
+   * @coversDefaultClass \Drupal\rules\Plugin\RulesAction\SystemPageRedirect
+   * @group rules_actions
+   */
+  class SystemPageRedirectTest extends RulesIntegrationTestBase {
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * The action to be tested.
+     *
+     * @var \Drupal\rules\Plugin\RulesAction\SystemPageRedirect
+     */
+    protected $action;
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\RequestStack
+     */
+    protected $request_stack;
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\Request
+     */
+    protected $current_request;
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\ParameterBag
```

Comment:
If this is a mock then it should also have |\Prophecy\Prophecy\ProphecyInterface, see the other action tests.